src/models/ComponentStyle.js

```diff
@@ -28,17 +60,37 @@ export default (nameGenerator: NameGenerator, flatten: Flattener, stringifyRules
      * Returns the hash to be injected on render()
      * */
     generateAndInjectStyles(executionContext: Object, styleSheet: StyleSheet) {
+      const { isStaticRules, isInjected, staticGeneratedClassName } = this
+
+      if (isStaticRules && staticGeneratedClassName && isInjected) {
+        return this.staticGeneratedClassName
+      }
+
+      const needsStaticName = isStaticRules && !staticGeneratedClassName
       const flatCSS = flatten(this.rules, executionContext)
       const hash = hashStr(this.componentId + flatCSS.join(''))
 
       const existingName = styleSheet.getName(hash)
-      if (existingName) return existingName
+      if (existingName) {
```

Comment:
Now that I'm seeing this, can you change this to `existingName !== undefined` please? :)